kolena/_utils/batched_load.py

```diff
@@ -79,6 +82,18 @@ def upload_data_frame_chunk(df_chunk: pd.DataFrame, load_uuid: str) -> None:
 DFType = TypeVar("DFType", bound=LoadableDataFrame)
 
 
+def upload_smart_chunk_data_frame(df: pd.DataFrame, uuid: str, rows: int = 1000) -> None:
```

Comment:
One idea is in `upload_data_frame_chunk`, where we are already doing the `to_parquet` and would have actual compressed size from `BytesIO` buffer, to split if the chunk size exceeds limit.

Reply:
Going to take a look at that and see how it works.